File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/inss/sobresalarios/OcorrenciaDeInssSalarioPagoOptimizerListSearchUnique.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios;

import br.jus.trt8.pjecalc.negocio.comum.OptimizerListSearchUnique;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.OcorrenciaDeInssSobreSalariosPagos;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.OcorrenciaInssUnique;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;

public class OcorrenciaDeInssSalarioPagoOptimizerListSearchUnique
extends OptimizerListSearchUnique<OcorrenciaInssUnique, OcorrenciaDeInssSobreSalariosPagos> {
    private Map<OcorrenciaInssUnique, OcorrenciaDeInssSobreSalariosPagos> map;

    @Override
    public OptimizerListSearchUnique<OcorrenciaInssUnique, OcorrenciaDeInssSobreSalariosPagos> init(Collection<OcorrenciaDeInssSobreSalariosPagos> collection) {
        this.map = new HashMap<OcorrenciaInssUnique, OcorrenciaDeInssSobreSalariosPagos>();
        OcorrenciaInssUnique key = null;
        for (OcorrenciaDeInssSobreSalariosPagos ocorrencia : collection) {
            key = new OcorrenciaInssUnique(ocorrencia);
            this.map.put(key, ocorrencia);
        }
        return this;
    }

    @Override
    public OcorrenciaDeInssSobreSalariosPagos search(OcorrenciaInssUnique key) {
        return this.map.get(key);
    }
}

File: src/lib/pjecalc/core/dominio/calculo/inss/sobresalarios/ocorrencia-de-inss-salario-pago-optimizer-list-search-unique.ts
/**
 * PJe-Calc v2.15.1 — OcorrenciaDeInssSalarioPagoOptimizerListSearchUnique
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.OcorrenciaDeInssSalarioPagoOptimizerListSearchUnique
 *
 * Ref Java: pjecalc-fonte/.../sobresalarios/OcorrenciaDeInssSalarioPagoOptimizerListSearchUnique.java
 */
import { OptimizerListSearchUnique } from '../../../../comum/optimizer-list-search-unique';
import { OcorrenciaInssUnique } from './ocorrencia-inss-unique';
import type { OcorrenciaDeInssSobreSalariosPagos } from './ocorrencia-de-inss-sobre-salarios-pagos';

export class OcorrenciaDeInssSalarioPagoOptimizerListSearchUnique extends OptimizerListSearchUnique<OcorrenciaInssUnique, OcorrenciaDeInssSobreSalariosPagos> {
  private map: Map<string, OcorrenciaDeInssSobreSalariosPagos> = new Map();

  init(collection: Iterable<OcorrenciaDeInssSobreSalariosPagos>): this {
    this.map = new Map();
    for (const ocorrencia of collection) {
      const key = new OcorrenciaInssUnique(ocorrencia);
      this.map.set(key.toKey(), ocorrencia);
    }
    return this;
  }

  search(key: OcorrenciaInssUnique): OcorrenciaDeInssSobreSalariosPagos | null {
    return this.map.get(key.toKey()) ?? null;
  }
}
